javascript-challenges.test.js

```diff
@@ -38,6 +106,61 @@ const hand3 = [5, 5, 5, 5, 4]
 const hand4 = [7, 2, 7, 2, 7]
 // Expected output: true
 
-// b) Create the function that makes the test pass.
+// // b) Create the function that makes the test pass.
+
+// // Pseudo code:
+
+// function to count occurrences of each element in an array
+const countHand = (hand) =>
+// Initialize an empty object to store counts
+// Iterate through each element in the array for num in arr
+// Update counts object with the current element
+// counts[num] = (counts[num] || 0) + 1
+// "counts" is are accumulator "num" is our currentValue
+// Set our initializer to 0
+  hand.reduce((counts, num) => ({ ...counts, [num]: (counts[num] || 0) + 1 }),{
+})
+// Function to check if an array represents a full house
+const isFullHouse = (hand) => {
+// Get the counts of occurrences for each element in the array
+  const counts = countHand(hand)
+//Create a variable to check if the array has 2 of the same integer value
+  const hasPair = Object.values(counts).includes(2)
+//Create a variable to check if the array has 3 of the same integer value
+  const hasThreeOfAKind = Object.values(counts).includes(3)
+//Return both variables to get a fullhouse
+  return hasPair && hasThreeOfAKind
+}
+
+// // const result1 = isFullHouse([5, 5, 5, 3, 3])
+
+// // Jest test
+// describe('isFullHouse', () => {
+//   it('should return a full house hand', () => {
+//     expect(hand1.isFullHouse).toEqual(true)
+//     expect(hand2.isFullHouse).toEqual(false)
+//     expect(hand3.isFullHouse).toEqual(false)
+//     expect(hand4.isFullHouse).toEqual(true)
+//   })
+// })
+
+// Function to count occurrences of each element in an array
+// const countHand = (hand) =>
+//   hand.reduce((counts, num) => ({ ...counts, [num]: (counts[num] || 0) + 1 }), {})
 
-// Pseudo code:
+// // Function to check if an array represents a full house
+// const isFullHouse = (hand) => {
+//   const counts = countHand(hand)
+//   const hasPair = Object.values(counts).includes(2)
+//   const hasThreeOfAKind = Object.values(counts).includes(3)
+//   return hasPair && hasThreeOfAKind
+// }
+// Jest test
+describe('isFullHouse', () => {
+  it('should return a full house', () => {
+    expect(isFullHouse(hand1)).toEqual(true)
+    expect(isFullHouse(hand2)).toEqual(false)
+    expect(isFullHouse(hand3)).toEqual(false)  
+    expect(isFullHouse(hand4)).toEqual(true)
+  })
+})
```

Comment:
I don't know if you used outside resources, but this code is fantastic (lines 147-166).  Remember that your test should come before your test.  Great job.